2-promises/solution.js

```diff
@@ -26,14 +26,33 @@ const id = yourRandomMethod() //third run
 
 function solution() {
     // YOUR SOLUTION GOES HERE
+    const lastName = require('./lastnames');
+    const firstName = require('./firstnames');
 
     // You generate your id value here
+    const interval = number => 
+        Math.floor(Math.random() * number);
 
-    // You call the lastnames method with your id
+    const possibles =  [
+        undefined, null, true, 'string',
+        interval(101) * (-1)**interval(2),
+        ...Array(5).fill(interval(5))
+    ];
 
-    // You call the firstname method
+    const id = possibles[interval(possibles.length)];   
 
-    // You log the fullname, or error, here
+    const fullName = {};
+    // You call the lastnames method with your id
+    lastName(id)
+        .then(lastName => {
+            fullName.lastName = lastName;
+            return firstName(lastName);
+        })
+        // You call the firstname method
+        .then(firstName => fullName.firstName = firstName)
+        // You log the fullname, or error, here
+        .then(_ => console.log(`${fullName.firstName} ${fullName.lastName}`))
```

Comment:
Here we can use the 2 constants created above

Reply:
ok, it is more better and more readable that way